Errors for Graph Type (#341)
neo4j/docs-status-codes: modules/ROOT/pages/errors/gql-errors/22NC5.adoc

= 22NC5

== Status description
error: data exception - graph type element not found. The `{ <<entityType>> }` element type referenced by `{ <<graphTypeReference>> }` does not exist.

== Explanation
This error occurs when a graph type definition references an element type (such as a node or relationship) using an alias that has not been defined, or by an identifying reference where there is no graph type element identified by that reference.

> If the referenced node or relationship type does not exist, the operation cannot proceed and this error is returned to indicate the missing element.

Reviewer: ```suggestion
If the referenced node or relationship type does not exist, the operation cannot proceed and will return an error to indicate the missing element.
```

== Example scenarios

=== Node element reference does not exist
For example, try to set a graph type with an identifying node element reference that does not exist as follows:

[source,cypher]
----
ALTER CURRENT GRAPH TYPE SET {
  (:Node =>)-[e:REL => { prop :: STRING }]->()
}
----

The query returns an error with GQLSTATUS 22NC5 and the status description:

[source]
----
error: data exception - graph type element not found. The node type element referenced by '(:Node =>)' does not exist.
----

> In this case, you can make the reference to be non-identifying, or you can define the node type element before this operation:

Reviewer: ```suggestion
In this case, you can make the reference non-identifying, or you can define the node type element before this operation:
```
